lib/puppet/type/posix_acl.rb

```diff
@@ -278,7 +278,7 @@ def generate
     if File.directory?(self[:path])
       Dir.chdir(self[:path]) do
         Dir['**/*'].each do |path|
-          paths << ::File.join(self[:path], path)
+          paths += ::File.join(self[:path], path)
```

Comment:
I'm not sure this is needed. `paths` is a [Set](https://ruby-doc.org/stdlib-2.7.1/libdoc/set/rdoc/Set.html), not a String. `<<` is an alias for [add](https://ruby-doc.org/stdlib-2.7.1/libdoc/set/rdoc/Set.html#method-i-add). Similar, `results` is an array where `<<` is also an add operation.